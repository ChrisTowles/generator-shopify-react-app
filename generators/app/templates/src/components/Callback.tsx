import * as React from "react";

class CallbackProps {
    errorMessage: string | null;
}

export function Callback(props: CallbackProps) {
    const error = props.errorMessage ? <div className="container__form">ERROR: {props.errorMessage}</div> : null;
    return (
        <main className="container" role="main">
            <header>
                <h1>Shopify App â€” Installation</h1>
                <p className="subhead">
                    <label htmlFor="shop">Redirecting, please wait...</label>
                </p>
            </header>

            {error}
        </main>
    );
}