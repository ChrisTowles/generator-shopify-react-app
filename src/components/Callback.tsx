import * as React from "react";

class CallbackProps {
    errorMessage: string | null;
}

export function Callback(props: CallbackProps) {
    const error = props.errorMessage ? <div className="alert error"><dl><dt>Error Alert</dt><dd>{props.errorMessage}</dd></dl></div> : null;
    return (
        <section className="zero-state full-width">
            <article>
                <h1>Shopify App â€” Installation</h1>
                {error}
            </article>
        </section>
    );
}