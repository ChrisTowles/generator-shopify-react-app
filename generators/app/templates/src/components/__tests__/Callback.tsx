import * as React from "react";
import * as renderer from "react-test-renderer";
import { Callback } from "../Callback";

test("Renders correctly with no error", () => {
  const login = <Callback errorMessage={null} loginUrl="/" />;

  const component = renderer.create(login);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Renders correctly with error", () => {
  const login = <Callback errorMessage={"Something went wrong"} loginUrl="/" />;

  const component = renderer.create(login);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
