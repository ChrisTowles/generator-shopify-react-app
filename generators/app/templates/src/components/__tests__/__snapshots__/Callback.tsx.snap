// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly with error 1`] = `
<main
  role="main"
  style={
    Object {
      "marginTop": "100px",
    }
  }
>
  <header
    style={
      Object {
        "marginBottom": "50px",
        "textAlign": "center",
      }
    }
  >
    <p
      className="Polaris-DisplayText Polaris-DisplayText--sizeExtraLarge"
    >
      Shopify App — Installation
    </p>
  </header>
  <div
    style={
      Object {
        "margin": "auto",
        "width": "400px",
      }
    }
  >
    <div
      aria-describedby="Banner1Content"
      aria-labelledby="Banner1Heading"
      aria-live="polite"
      className="Polaris-Banner Polaris-Banner--statusCritical"
      onMouseUp={[Function]}
      role="alert"
      tabIndex={0}
    >
      <div
        className="Polaris-Banner__Ribbon"
      >
        <span
          aria-label={undefined}
          className="Polaris-Icon Polaris-Icon--colorRedDark Polaris-Icon--hasBackdrop"
        >
          <svg
            className="Polaris-Icon__Svg"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<g  fill-rule=\\"evenodd\\"><circle fill=\\"currentColor\\" cx=\\"10\\" cy=\\"10\\" r=\\"9\\"/><path d=\\"M2 10c0-1.846.635-3.543 1.688-4.897l11.209 11.209A7.954 7.954 0 0 1 10 18c-4.411 0-8-3.589-8-8m14.312 4.897L5.103 3.688A7.954 7.954 0 0 1 10 2c4.411 0 8 3.589 8 8a7.952 7.952 0 0 1-1.688 4.897M0 10c0 5.514 4.486 10 10 10s10-4.486 10-10S15.514 0 10 0 0 4.486 0 10\\" /></g>",
              }
            }
            viewBox="0 0 20 20"
          />
        </span>
      </div>
      <div>
        <div
          className="Polaris-Banner__Heading"
          id="Banner1Heading"
        >
          <p
            className="Polaris-Heading"
          >
            Authentication Error
          </p>
        </div>
        <div
          className="Polaris-Banner__Content"
          id="Banner1Content"
        >
          <p>
            Something went wrong
          </p>
          <div
            className="Polaris-Banner__Actions"
          >
            <div
              className="Polaris-ButtonGroup"
            >
              <div
                className="Polaris-ButtonGroup__Item"
                onBlur={[Function]}
                onFocus={[Function]}
              >
                <a
                  aria-label={undefined}
                  className="Polaris-Button Polaris-Button--outline"
                  data-polaris-unstyled={true}
                  disabled={undefined}
                  href="/"
                  onBlur={undefined}
                  onClick={undefined}
                  onFocus={undefined}
                  onMouseUp={[Function]}
                  rel={undefined}
                  target={undefined}
                >
                  <span
                    className="Polaris-Button__Content"
                  >
                    <span>
                      Try again
                    </span>
                  </span>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</main>
`;

exports[`Renders correctly with no error 1`] = `
<main
  role="main"
  style={
    Object {
      "marginTop": "100px",
    }
  }
>
  <header
    style={
      Object {
        "marginBottom": "50px",
        "textAlign": "center",
      }
    }
  >
    <p
      className="Polaris-DisplayText Polaris-DisplayText--sizeExtraLarge"
    >
      Shopify App — Installation
    </p>
  </header>
  <div
    style={
      Object {
        "margin": "auto",
        "width": "400px",
      }
    }
  >
    <p>
      Redirecting, please wait...
    </p>
  </div>
</main>
`;
